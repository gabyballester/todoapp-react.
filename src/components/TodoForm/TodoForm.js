import { useState, useEffect, useRef } from 'react';

export default function TodoForm(props) {
  const [input, setInput] = useState(props.edit ? props.edit.value
    : '');

  const inputRef = useRef(null);

  useEffect(() => {
    inputRef.current.focus()
  })

  const handleChange = e => {
    setInput(e.target.value)
  }

  const handleSubmit = e => {
    e.preventDefault();

    props.onSubmit({
      id: Math.floor(Math.random() * 10000),
      text: input
    })

    setInput('')
  }


  return (
    <form className="todo-form" onSubmit={handleSubmit}>
      <input
        data-testid="input"
        type="text"
        placeholder={props.edit ? "Edita la tarea" : "Añade la tarea"}
        value={input}
        name="text"
        className={props.edit ? "todo-input todo-input-edit" : "todo-input"}
        onChange={handleChange}
        ref={inputRef}
      />
      <button className="todo-button">{props.edit ? "Editar" : "Añadir"}</button>
    </form>
  )
}
