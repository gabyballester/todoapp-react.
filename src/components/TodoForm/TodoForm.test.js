import React from 'react';
import ReactDOM from 'react-dom';
import TodoForm from './TodoForm';
import { render,cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  const form = document.createElement("form");
  ReactDOM.render(<TodoForm></TodoForm>, form)
})

it("renders input correctly", () => {
  const {getByTestId} = render(<TodoForm></TodoForm>)
  expect(getByTestId('input')).toHaveTextContent('');
})

it("matches snapshot", ()=>{
  const tree = renderer.create(<TodoForm></TodoForm>).toJSON();
  expect(tree).toMatchSnapshot();
})