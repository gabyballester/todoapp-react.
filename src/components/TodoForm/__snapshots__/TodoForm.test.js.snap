// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className="todo-form"
  onSubmit={[Function]}
>
  <input
    className="todo-input"
    data-testid="input"
    name="text"
    onChange={[Function]}
    placeholder="Añade la tarea"
    type="text"
    value=""
  />
  <button
    className="todo-button"
  >
    Añadir
  </button>
</form>
`;
